lib/preview/httpsRedirectingServer.js

```diff
@@ -0,0 +1,85 @@
+const http = require('http');
+const https = require('https');
+const net = require('net');
+const { unlinkSync } = require('fs');
+const { silenceConsoleWhile } = require('./previewUtils');
+
+const createCert = async (domainsToProxy) => {
+  const additionalMkcertHosts = domainsToProxy
+    .map(proxyDomain => [
+      `${proxyDomain.domain}.localhost`,
+      `${proxyDomain.domain}.hslocal.net`
+    ])
+    .flat();
+  const hosts = ['localhost', 'hslocal.net', ...additionalMkcertHosts];
+  const { createCertificate } = await import('mkcert-cli');
+  const { key, cert } = await silenceConsoleWhile(createCertificate, {
+    keyFilePath: `${__dirname}/key.pem`,
+    certFilePath: `${__dirname}/cert.pem`
+  }, hosts);
+  unlinkSync(`${__dirname}/key.pem`);
+  unlinkSync(`${__dirname}/cert.pem`);
+  return { key, cert };
+}
```

Comment:
`cms-js-platform` uses `vite-plugin-mkcert`. This is handy cause it takes care of everything it seems...generates the CA and key/cert, and does all the system trust config stuff under the hood. Unless we want to try to incorporate vite for this we need an alternate - I looked at
- using the original `mkcert` tool on its own (written in Go so we'd need to do some surgery to incorporate it)
 - digging into the internals of `vite-plugin-mkcert` and just using the bits we need. Maybe fork it and make some changes or borrow from it. But seemed heavily involved, a lot of work needs to be done to locate `mkcert` installs and downloads and such.
 - `mkcert` on npm, which generates CA and key/certs but does not handle the system trust stuff, you're expected to do that on your own
 - This, `mkcert-cli`, which is explicitly meant to be `vite-plugin-mkcert` extracted away from vite, and handle all that stuff. 
 
The only quirks are that there's no option to since logging (hence the wrapper here to temporarily overwrite `console` while it runs, and that there's no option to not store the key/cert and try to read from it. So I just delete them after.